packages/contracts/test/voting/admin-address-setup.ts

```diff
@@ -0,0 +1,184 @@
+import {expect} from 'chai';
+import {ethers} from 'hardhat';
+
+import {AdminAddressSetup} from '../../typechain';
+import {customError} from '../test-utils/custom-error-helper';
+import {deployNewDAO} from '../test-utils/dao';
+import {getInterfaceID} from '../test-utils/interfaces';
+
+enum Op {
+  Grant,
+  Revoke,
+  Freeze,
+  GrantWithOracle,
+}
+
+const abiCoder = ethers.utils.defaultAbiCoder;
+const AddressZero = ethers.constants.AddressZero;
+const EMPTY_DATA = '0x';
+
+// Permissions
+const ADMIN_EXECUTE_PERMISSION_ID = ethers.utils.id('ADMIN_EXECUTE_PERMISSION');
+const EXECUTE_PERMISSION_ID = ethers.utils.id('EXECUTE_PERMISSION');
+
+describe('AdminAddressSetup', function () {
+  let ownerAddress: string;
+  let signers: any;
+  let adminAddressSetup: AdminAddressSetup;
+  let implementationAddress: string;
+  let targetDao: any;
+  let minimum_data: any;
+
+  before(async () => {
+    signers = await ethers.getSigners();
+    ownerAddress = await signers[0].getAddress();
+    targetDao = await deployNewDAO(ownerAddress);
+
+    minimum_data = abiCoder.encode(['address'], [ownerAddress]);
+
+    const AdminAddressSetup = await ethers.getContractFactory(
+      'AdminAddressSetup'
+    );
+    adminAddressSetup = await AdminAddressSetup.deploy();
+
+    implementationAddress = await adminAddressSetup.getImplementationAddress();
+  });
+
+  it('creates admin address base with the correct interface', async () => {
+    const factory = await ethers.getContractFactory('AdminAddress');
+    const adminAddressContract = factory.attach(implementationAddress);
+
+    const iface = new ethers.utils.Interface([
+      'function initialize(address  _dao)',
+      'function executeProposal(bytes _proposalMetadata, tuple(address,uint256,bytes)[] _actions)',
+    ]);
+
+    expect(
+      await adminAddressContract.supportsInterface(getInterfaceID(iface))
+    ).to.be.eq(true);
+  });
+
+  describe('prepareInstallation', async () => {
+    it('correctly returns prepare installation data abi', async () => {
+      // Human-Readable Abi of data param of `prepareInstallation`.
+      const dataHRABI = '(address admin)';
+
+      expect(await adminAddressSetup.prepareInstallationDataABI()).to.be.eq(
+        dataHRABI
+      );
+    });
+
+    it('fails if data is empty, or not of minimum length', async () => {
+      await expect(
+        adminAddressSetup.prepareInstallation(targetDao.address, EMPTY_DATA)
+      ).to.be.reverted;
+
+      await expect(
+        adminAddressSetup.prepareInstallation(
+          targetDao.address,
+          minimum_data.substring(0, minimum_data.length - 1)
+        )
+      ).to.be.reverted;
```

Comment:
Don't we need `()` here for it to be executed?

Reply:
i don't think so,
if you use `**.to.not.be.reverted` you see you will get: `AssertionError: Expected transaction NOT to be reverted`